astropy/utils/data.py

```diff
@@ -280,9 +280,8 @@ def get_readable_fileobj(name_or_obj, encoding=None, cache=False,
         except ImportError:
             for fd in close_fds:
                 fd.close()
-            raise ValueError(
-                ".bz2 format files are not supported since the Python "
-                "interpreter does not include the bz2 module")
+            raise ModuleNotFoundError(
```

Comment:
While this seems appropriate and unifies the errors encountered when builtin modules are missing this PR is marked as bugfix so we shouldn't change already existing API (like changing the exception).

Reply:
OK, if the rules on API changes don't allow it, I guess it's better to live with the inconsistent exceptions for a while.